src/com/itacademy/eshop/services/ShopService.java

```diff
@@ -1,29 +1,44 @@
 package com.itacademy.eshop.services;
 
+import com.itacademy.eshop.product.Discount;
 import com.itacademy.eshop.product.Product;
+import com.itacademy.eshop.product.types.Category;
+import com.itacademy.eshop.product.types.State;
 import com.itacademy.eshop.shop.Eshop;
 
 import java.util.ArrayList;
+import java.util.List;
 
 public class ShopService {
     /*** This class is used for creating new e-shop object and initializing it with some products.
      */
 
     public Eshop createShop() {
+
         /** Should create new e-shop object and initialize it with some products.
          * Should return e-shop object.
          */
-       ArrayList<Product> products = createProducts();
-        throw new UnsupportedOperationException("createShop() is not implemented yet");
+        ArrayList<Product> products = createProducts();
+        return new Eshop("Rut-shop", products);   // returning in a single line
     }
 
     private ArrayList<Product> createProducts() {
+
+        ArrayList<Product> arrayList = new ArrayList<>();
```

Comment:
Have a bit more descriptive name for the parameter. Now it is hard to understand what does the list contain. by having something like products or productList we will know that ok, there is a collection of products here.